src/librustc/lint/context.rs

```diff
@@ -462,7 +462,13 @@ pub fn raw_struct_lint<'a>(sess: &'a Session,
 
     if let Some(span) = def {
         let explanation = "lint level defined here";
-        err.span_note(span, &explanation);
+        let span_explanation = (span, explanation.to_owned());
+        let already_noted: bool = sess.one_time_diagnostics.borrow()
```

Comment:
I think it'd be nice to offer a method that encapsulates this "check-and-add" pattern rather than modifying the field directly (in general, I think it's best practice to hide a `RefCell` behind methods in any case, but in particular here).

For example, this code might look like:

```rust
err.span_note_once(span, explanation);
```

and then this method would do the check.